WF-295: Write a migration guide (#19)
spatialos/FlexibleProjectExample: docs/migration-guide/configs-conversion-guide.md

| `componentDelivery` | `bridge/component_delivery` field in your worker configuration file |
| `componentSettings` | `bridge/component_settings` field in your worker configuration file |
| `permissions` | Similar to the `workers/<my-worker-type>/permissions` field in your [launch configuration file](https://docs.improbable.io/reference/latest/shared/reference/file-formats/launch-config#launch-configuration-file), with the difference that the permissions property here is a single [permission object](https://docs.improbable.io/reference/latest/shared/worker-configuration/permissions#worker-permissions) rather than an array |
| `uploadConfiguration/<platform>/localBuildDirectory` (client-worker only) | No equivalent - refer to [the documentation](https://github.com/spatialos/FlexibleProjectExample/blob/master/docs/reference/client-worker-configuration.md) for information on how to populate this field |
| `launchConfiguration/<platform>/localBuildDirectory` (server-worker only) | No equivalent - refer to [the documentation](https://github.com/spatialos/FlexibleProjectExample/blob/master/docs/reference/client-worker-configuration.md) for information on how to populate this field |

> | `launchConfiguration/<localDeployment, cloudDeployment>/<platform>/<command, arguments>` (server-worker only) | `managed/<platform>/<command, arguments>` field in you worker configuration file |

Reviewer: ```suggestion
| `launchConfiguration/<localDeployment, cloudDeployment>/<platform>/<command, arguments>` (server-worker only) | `managed/<platform>/<command, arguments>` field in your worker configuration file |
```

Author: Done.

**Notes:**

> - The `build` field in the CPL worker configuration is obsolete and does not have an equivalent in the FPL. In the FPL, you fully manage the build process of a worker. For more information, see [Building a worker executable in the FPL](../build-process/worker-build-process.md).

Reviewer: ```suggestion
- The `build` field in the CPL worker configuration file is obsolete and does not have an equivalent in the FPL. In the FPL, you fully manage the build process of a worker. For more information, see [Building a worker executable in the FPL](../build-process/worker-build-process.md).
```

Author: Done.

> - The `external` field in the CPL worker configuration is obsolete and does not have an equivalent in the FPL. In the FPL, you need to start a worker manually instead of using `spatial local worker launch`.

Reviewer: ```suggestion
- The `external` field in the CPL worker configuration file is obsolete and does not have an equivalent in the FPL. In the FPL, you need to start a worker manually instead of using `spatial local worker launch`.
```

Author: Done.

| --- | --- |
| `template` | `template` field in your launch configuration file |
| `dimensionsInWorldUnits/<x, z>` | `world/dimensions/<xMeters, zMeters>` field in your launch configuration file |
| `loadBalancing` | `load_balancing` field in your launch configuration file |
| `snapshot` | No equivalent - refer to [the documentation](https://github.com/spatialos/FlexibleProjectExample/blob/master/docs/reference/deployment-launch-configuration.md) for information on how to populate this field |
| `takeSnapshotIntervalSeconds` | `world/snapshots/snapshot_write_period_seconds` field in your launch configuration file |
| `startDeploymentFromSnapshotFile`| No equivalent - refer to [the documentation](https://github.com/spatialos/FlexibleProjectExample/blob/master/docs/reference/deployment-launch-configuration.md) for information on how to populate this field |
| `workerFlags/<elem>/<workerType, flags>` | `workers/<elem>/<worker_type, flags>` field in your launch configuration file |
| `streamingQueryInterval` | `world/streaming_query_interval` field in your launch configuration file |
| `runtimeFlags` | `world/legacy_flag` field in your launch configuration file |

**Notes:**

> - The `world/legacy_javaparams` field in the CPL launch configuration is obsolete and does not have an equivalent in the FPL.

Reviewer: ```suggestion
- The `world/legacy_javaparams` field in the CPL launch configuration file is obsolete and does not have an equivalent in the FPL.
```

Author: Done.

> - The `world/chunk_edge_length_meters` field in CPL launch configuration is obsolete and does not have an equivalent in the FPL. In the FPL, this parameter has a fixed value of 100. You might need to adapt the [entity interest](https://docs.improbable.io/reference/latest/shared/worker-configuration/bridge-config#entity-interest) configuration of your workers to account for the change in this value when you migrate a project from the CPL to the FPL.

Reviewer: ```suggestion
- The `world/chunk_edge_length_meters` field in the CPL launch configuration file is obsolete and does not have an equivalent in the FPL. In the FPL, this parameter has a fixed value of 100. You might need to adapt the [entity interest](https://docs.improbable.io/reference/latest/shared/worker-configuration/bridge-config#entity-interest) configuration of your workers to account for the change in this value when you migrate a project from the CPL to the FPL.
```

Author: Done.

Below is a summary of how to populate the fields of an FPL project configuration file based on the state of your existing CPL project:

| FPL format | CPL format |
| --- | --- |
| `configurationVersion` | No equivalent - refer to [the documentation](https://github.com/spatialos/FlexibleProjectExample/blob/master/docs/reference/project-configuration.md) for information on how to populate this field |
| `projectName` | `name` field in your project definition file |
| `schemaDescriptor` | No equivalent - refer to [the documentation](https://github.com/spatialos/FlexibleProjectExample/blob/master/docs/reference/project-configuration.md) for information on how to populate this field |
| `clientWorkers` | No equivalent - refer to [the documentation](https://github.com/spatialos/FlexibleProjectExample/blob/master/docs/reference/project-configuration.md) for information on how to populate this field |
| `serverWorkers` | No equivalent - refer to [the documentation](https://github.com/spatialos/FlexibleProjectExample/blob/master/docs/reference/project-configuration.md) for information on how to populate this field |

**Notes:**

> - The `project_version` field in the CPL project definition is obsolete and does not have an equivalent in the FPL. Please keep track of your project/built versions separately.

Reviewer: ```suggestion
- The `project_version` field in the CPL project definition file is obsolete and does not have an equivalent in the FPL. Please keep track of your project/built versions separately.
```

Author: Done.

> - The `sdk_version` field in the CPL project definition is obsolete and does not have an equivalent in the FPL. There is no clearly defined SDK version for an FPL project anymore. For more information, please see [Choose an SDK version](../build-process/worker-build-process.md#choose-an-sdk-version).

Reviewer: ```suggestion
- The `sdk_version` field in the CPL project definition file is obsolete and does not have an equivalent in the FPL. There is no longer a clearly defined SDK version for an FPL project. For more information, please see [Choose an SDK version](../build-process/worker-build-process.md#choose-an-sdk-version).
```

Author: Done.

> - The `dependencies` field in the CPL is obsolete and does not have an equivalent in the FPL. You need to manage your dependencies manually as part of your build process. For more information, please see [Download the relevant worker SDK libraries](../build-process/worker-build-process.md#download-the-relevant-worker-sdk-libraries).

Reviewer: ```suggestion
- The `dependencies` field in the CPL project definition file is obsolete and does not have an equivalent in the FPL. You need to manage your dependencies manually as part of your build process. For more information, please see [Download the relevant worker SDK libraries](../build-process/worker-build-process.md#download-the-relevant-worker-sdk-libraries).
```

Author: Done.
